### PR TITLE
Add 1.18 support

### LINKED ISSUE #66: Minecraft 1.18 not supported
Currently Minecraft 1.18 is not supported, because the fields and methods names in the NMS classes are now obfuscated.

This doesn't work:
- Actionbars
- Titles
- EntitySound
- Books

Debug output:
```
15:52:10] [Server thread/INFO]: Skipped facet: net.kyori.adventure.platform.bukkit.BukkitAudience$$Lambda$4770/0x0000000801aa5440
[15:52:10] [Server thread/INFO]: Skipped facet: net.kyori.adventure.platform.bukkit.CraftBukkitFacet$Chat@37e6578
[15:52:10] [Server thread/INFO]: Added facet: net.kyori.adventure.platform.bukkit.BukkitFacet$Chat@bd809f5
[15:52:10] [Server thread/INFO]: Skipped facet: net.kyori.adventure.platform.bukkit.BukkitAudience$$Lambda$4779/0x0000000801aaa220
[15:52:10] [Server thread/INFO]: Skipped facet: net.kyori.adventure.platform.bukkit.BukkitAudience$$Lambda$4780/0x0000000801aaa440
[15:52:10] [Server thread/INFO]: Skipped facet: net.kyori.adventure.platform.bukkit.CraftBukkitFacet$ActionBar_1_17@50962d45
[15:52:10] [Server thread/INFO]: Skipped facet: net.kyori.adventure.platform.bukkit.CraftBukkitFacet$ActionBar@2e664cf3
[15:52:10] [Server thread/INFO]: Skipped facet: net.kyori.adventure.platform.bukkit.CraftBukkitFacet$ActionBarLegacy@167e02b3
[15:52:10] [Server thread/INFO]: Skipped facet: net.kyori.adventure.platform.bukkit.BukkitAudience$$Lambda$4784/0x0000000801aabaa8
[15:52:10] [Server thread/INFO]: Skipped facet: net.kyori.adventure.platform.bukkit.CraftBukkitFacet$Title_1_17@7f681954
[15:52:10] [Server thread/INFO]: Skipped facet: net.kyori.adventure.platform.bukkit.CraftBukkitFacet$Title@6b4d5d2a
[15:52:10] [Server thread/INFO]: Added facet: net.kyori.adventure.platform.bukkit.BukkitFacet$SoundWithCategory@1681f8b3
[15:52:10] [Server thread/INFO]: Added facet: net.kyori.adventure.platform.bukkit.BukkitFacet$Sound@ad9d6c
[15:52:10] [Server thread/WARN]: Failed to initialize EntitySound CraftBukkit facet
java.lang.NullPointerException: Cannot invoke "java.lang.invoke.MethodHandle.invoke(Object, Object)" because "net.kyori.adventure.platform.bukkit.CraftBukkitFacet$EntitySound.REGISTRY_GET_OPTIONAL" is null
	at net.kyori.adventure.platform.bukkit.CraftBukkitFacet$EntitySound.<clinit>(CraftBukkitFacet.java:447) ~[?:?]
	at net.kyori.adventure.platform.bukkit.BukkitAudience.lambda$static$13(BukkitAudience.java:70) ~[?:?]
	at net.kyori.adventure.platform.facet.Facet.of(Facet.java:71) ~[?:?]
	at net.kyori.adventure.platform.bukkit.BukkitAudience.<clinit>(BukkitAudience.java:69) ~[?:?]
	at net.kyori.adventure.platform.bukkit.BukkitAudiencesImpl.createAudience(BukkitAudiencesImpl.java:137) ~[?:?]
	at net.kyori.adventure.platform.bukkit.BukkitAudiencesImpl.createAudience(BukkitAudiencesImpl.java:74) ~[?:?]
	at net.kyori.adventure.platform.facet.FacetAudienceProvider.<init>(FacetAudienceProvider.java:99) ~[?:?]
	at net.kyori.adventure.platform.bukkit.BukkitAudiencesImpl.<init>(BukkitAudiencesImpl.java:95) ~[?:?]
	at net.kyori.adventure.platform.bukkit.BukkitAudiencesImpl$Builder.lambda$build$1(BukkitAudiencesImpl.java:168) ~[?:?]
	at java.util.HashMap.computeIfAbsent(HashMap.java:1220) ~[?:?]
	at java.util.Collections$SynchronizedMap.computeIfAbsent(Collections.java:2760) ~[?:?]
	at net.kyori.adventure.platform.bukkit.BukkitAudiencesImpl$Builder.build(BukkitAudiencesImpl.java:168) ~[?:?]
	at net.kyori.adventure.platform.bukkit.BukkitAudiencesImpl.instanceFor(BukkitAudiencesImpl.java:89) ~[?:?]
	at net.kyori.adventure.platform.bukkit.BukkitAudiences.create(BukkitAudiences.java:53) ~[?:?]
	at de.cytooxien.internationalization.spigot.manager.SpigotMessageManagerImpl.<init>(SpigotMessageManagerImpl.java:43) ~[i18n-spigot.jar:?]
	at de.cytooxien.internationalization.spigot.InternationalizationPlugin.onEnable(InternationalizationPlugin.java:36) ~[i18n-spigot.jar:?]
	at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:264) ~[paper-api-1.18-rc3-R0.1-SNAPSHOT.jar:?]
	at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:370) ~[paper-api-1.18-rc3-R0.1-SNAPSHOT.jar:?]
	at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:500) ~[paper-api-1.18-rc3-R0.1-SNAPSHOT.jar:?]
	at org.bukkit.craftbukkit.v1_18_R1.CraftServer.enablePlugin(CraftServer.java:565) ~[spigot.jar:git-Paper-"7db994d"]
	at org.bukkit.craftbukkit.v1_18_R1.CraftServer.enablePlugins(CraftServer.java:479) ~[spigot.jar:git-Paper-"7db994d"]
	at net.minecraft.server.MinecraftServer.loadWorld0(MinecraftServer.java:726) ~[spigot.jar:git-Paper-"7db994d"]
	at net.minecraft.server.MinecraftServer.loadLevel(MinecraftServer.java:501) ~[spigot.jar:git-Paper-"7db994d"]
	at net.minecraft.server.dedicated.DedicatedServer.initServer(DedicatedServer.java:313) ~[spigot.jar:git-Paper-"7db994d"]
	at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1201) ~[spigot.jar:git-Paper-"7db994d"]
	at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:316) ~[spigot.jar:git-Paper-"7db994d"]
	at java.lang.Thread.run(Thread.java:833) ~[?:?]
[15:52:10] [Server thread/INFO]: Skipped facet: net.kyori.adventure.platform.bukkit.CraftBukkitFacet$EntitySound@1672d3cd
[15:52:10] [Server thread/INFO]: Skipped facet: net.kyori.adventure.platform.bukkit.CraftBukkitFacet$BookPost1_13@6864f73a
[15:52:10] [Server thread/INFO]: Skipped facet: net.kyori.adventure.platform.bukkit.CraftBukkitFacet$Book1_13@3707b542
[15:52:10] [Server thread/INFO]: Skipped facet: net.kyori.adventure.platform.bukkit.CraftBukkitFacet$BookPre1_13@dcabb5f
[15:52:10] [Server thread/INFO]: Skipped facet: net.kyori.adventure.platform.viaversion.ViaFacet$BossBar$Builder@35b1d8a2
[15:52:10] [Server thread/INFO]: Skipped facet: net.kyori.adventure.platform.viaversion.ViaFacet$BossBar$Builder1_9_To_1_15@1d4fca84
[15:52:10] [Server thread/INFO]: Skipped facet: net.kyori.adventure.platform.bukkit.CraftBukkitFacet$BossBar$Builder@76aa21e3
[15:52:10] [Server thread/INFO]: Added facet: net.kyori.adventure.platform.bukkit.BukkitFacet$BossBarBuilder@5be363bb
[15:52:10] [Server thread/INFO]: Skipped facet: net.kyori.adventure.platform.bukkit.CraftBukkitFacet$BossBarWither$Builder@4a3ddfc5
[15:52:10] [Server thread/INFO]: Skipped facet: net.kyori.adventure.platform.viaversion.ViaFacet$TabList@5e39a7d7
[15:52:10] [Server thread/INFO]: Skipped facet: net.kyori.adventure.platform.bukkit.PaperFacet$TabList@7bd3f6ad
[15:52:10] [Server thread/INFO]: Skipped facet: net.kyori.adventure.platform.bukkit.CraftBukkitFacet$TabList@2ec756eb
[15:52:10] [Server thread/INFO]: Added facet: net.kyori.adventure.platform.bukkit.BukkitFacet$TabList@7ae88a6b
[15:52:10] [Server thread/INFO]: Added facet: net.kyori.adventure.platform.bukkit.BukkitFacet$CommandSenderPointers@722a2e6a
[15:52:10] [Server thread/INFO]: Added facet: net.kyori.adventure.platform.bukkit.BukkitFacet$ConsoleCommandSenderPointers@23269398
[15:52:10] [Server thread/INFO]: Added facet: net.kyori.adventure.platform.bukkit.BukkitFacet$PlayerPointers@79647d02
[15:52:10] [Server thread/INFO]: Selected facet: net.kyori.adventure.platform.bukkit.BukkitFacet$Chat@bd809f5 for com.destroystokyo.paper.console.TerminalConsoleCommandSender@6ca5e5ce
```

### PR DESCRIPTION
**Pull request by Foorcee**
Fixes #66 

```
[18:01:54] [Server thread/INFO]: Skipped facet: net.kyori.adventure.platform.bukkit.BukkitAudience$$Lambda$4769/0x0000000801aa9440
[18:01:54] [Server thread/INFO]: Added facet: net.kyori.adventure.platform.bukkit.CraftBukkitFacet$Chat@374caafe
[18:01:54] [Server thread/INFO]: Added facet: net.kyori.adventure.platform.bukkit.BukkitFacet$Chat@289b4bff
[18:01:54] [Server thread/INFO]: Skipped facet: net.kyori.adventure.platform.bukkit.BukkitAudience$$Lambda$4782/0x0000000801ab0298
[18:01:54] [Server thread/INFO]: Skipped facet: net.kyori.adventure.platform.bukkit.BukkitAudience$$Lambda$4783/0x0000000801ab04b8
[18:01:54] [Server thread/INFO]: Added facet: net.kyori.adventure.platform.bukkit.CraftBukkitFacet$ActionBar_1_17@167e02b3
[18:01:54] [Server thread/INFO]: Skipped facet: net.kyori.adventure.platform.bukkit.CraftBukkitFacet$ActionBar@454f6930
[18:01:54] [Server thread/INFO]: Skipped facet: net.kyori.adventure.platform.bukkit.CraftBukkitFacet$ActionBarLegacy@2641708a
[18:01:54] [Server thread/INFO]: Skipped facet: net.kyori.adventure.platform.bukkit.BukkitAudience$$Lambda$4787/0x0000000801ab1b20
[18:01:54] [Server thread/INFO]: Added facet: net.kyori.adventure.platform.bukkit.CraftBukkitFacet$Title_1_17@10199ab2
[18:01:54] [Server thread/INFO]: Skipped facet: net.kyori.adventure.platform.bukkit.CraftBukkitFacet$Title@7430cf7d
[18:01:54] [Server thread/INFO]: Added facet: net.kyori.adventure.platform.bukkit.BukkitFacet$SoundWithCategory@26cce560
[18:01:54] [Server thread/INFO]: Added facet: net.kyori.adventure.platform.bukkit.BukkitFacet$Sound@60864b19
[18:01:54] [Server thread/INFO]: Added facet: net.kyori.adventure.platform.bukkit.CraftBukkitFacet$EntitySound@1d9c582c
[18:01:54] [Server thread/INFO]: Added facet: net.kyori.adventure.platform.bukkit.CraftBukkitFacet$BookPost1_13@35b1d8a2
[18:01:54] [Server thread/INFO]: Skipped facet: net.kyori.adventure.platform.bukkit.CraftBukkitFacet$Book1_13@1d4fca84
[18:01:54] [Server thread/INFO]: Skipped facet: net.kyori.adventure.platform.bukkit.CraftBukkitFacet$BookPre1_13@76aa21e3
[18:01:54] [Server thread/INFO]: Skipped facet: net.kyori.adventure.platform.viaversion.ViaFacet$BossBar$Builder@1b8aa1f0
[18:01:54] [Server thread/INFO]: Skipped facet: net.kyori.adventure.platform.viaversion.ViaFacet$BossBar$Builder1_9_To_1_15@100402ac
[18:01:54] [Server thread/INFO]: Skipped facet: net.kyori.adventure.platform.bukkit.CraftBukkitFacet$BossBar$Builder@7bd3f6ad
[18:01:54] [Server thread/INFO]: Added facet: net.kyori.adventure.platform.bukkit.BukkitFacet$BossBarBuilder@6e0cf94f
[18:01:54] [Server thread/INFO]: Added facet: net.kyori.adventure.platform.bukkit.CraftBukkitFacet$BossBarWither$Builder@c1f7adf
[18:01:54] [Server thread/INFO]: Skipped facet: net.kyori.adventure.platform.viaversion.ViaFacet$TabList@23269398
[18:01:54] [Server thread/INFO]: Added facet: net.kyori.adventure.platform.bukkit.PaperFacet$TabList@47f7bf31
[18:01:54] [Server thread/INFO]: Added facet: net.kyori.adventure.platform.bukkit.CraftBukkitFacet$TabList@72f21dfd
[18:01:54] [Server thread/INFO]: Added facet: net.kyori.adventure.platform.bukkit.BukkitFacet$TabList@b7b3251
[18:01:54] [Server thread/INFO]: Added facet: net.kyori.adventure.platform.bukkit.BukkitFacet$CommandSenderPointers@4368c7b2
[18:01:54] [Server thread/INFO]: Added facet: net.kyori.adventure.platform.bukkit.BukkitFacet$ConsoleCommandSenderPointers@7299b553
[18:01:54] [Server thread/INFO]: Added facet: net.kyori.adventure.platform.bukkit.BukkitFacet$PlayerPointers@7caa8bdc
```

### CODE REVIEW

File: platform-bukkit/src/main/java/net/kyori/adventure/platform/bukkit/MinecraftReflection.java [hunk 1]
```diff
@@ -141,6 +142,65 @@ public static boolean hasClass(final @NotNull String... classNames) {
...
+              || !Arrays.equals(method.getParameterTypes(), parameterClasses)) continue;
+      try {
+        return LOOKUP.findVirtual(holderClass, method.getName(), MethodType.methodType(returnClass, parameterClasses));
+      } catch (final NoSuchMethodException | IllegalAccessException e) {
+      }
+    }
+    return null;
+  }
+
+  private static boolean matchModifiers(final int modifier, final @Nullable int[] modifiers) {
```
Comment (SirYwell) [inline platform-bukkit/src/main/java/net/kyori/adventure/platform/bukkit/MinecraftReflection.java:194]: Is there a specific reason why `methodModifiers` is an array? Modifiers are typically represented as bitset in one int.

Comment (astei) [reply platform-bukkit/src/main/java/net/kyori/adventure/platform/bukkit/MinecraftReflection.java:194]: Agreed. It's simple to verify whether a bit set contains any of a modified set of items: `bitSet & (FLAG_1 | FLAG_2) != 0`. Consider using this pattern instead.

**Comment by AlessioGr** [comment]
Hover & click messages did not work with 1.18 on my plugin. I tried out this PR, and now it works again. Thank you!

**Comment by Phoenix616** [comment]
I mean I don't want to be a stickler for method naming in other people's projects but it seems odd to name those methods `searchMethod` while all others are named way more optimistically with the `findXyz` pattern 🙈

**Comment by MiniDigger** [comment]
love the renovate notifications, look at all these people not upgrading to platforms that natively support adventure, s m h

**Comment by AlessioGr** [comment]
> love the renovate notifications, look at all these people not upgrading to platforms that natively support adventure, s m h

Oh, I would love to switch to a native platform once https://www.spigotmc.org/resources/ is dead and not used by 90% of people anymore 🥲

**Comment by MiniDigger** [comment]
ssssssh, am working on it! 😂

**Pull request merged**